src/main.cpp

```diff
@@ -1,9 +1,133 @@
-#include <cstdlib>
+#include <algorithm>
+#include <iostream>
+#include <iterator>
 #include <iostream>
+#include <cstdlib>
+#include <fstream>
+#include <sstream>
+#include <iomanip>
+#include <string>
+#include <vector>
+using namespace std;
+
+// Global Variables & Functions
+vector<string> all_types = {"PROGRAM", "VAR", "END_VAR", "END_PROGRAM", "BOOL", "TRUE", "FALSE"}; //keywords
+vector<string> operators = {"<", ">", "+", "-", "%", ":="}; // operators
+vector<string> all_sc = {"{", "}", "[", "]", ";", ":"}; // special characters
+void tokenizerFunc (string sample_input); // main lexer function
+typedef struct Token{ // where/how the keywords are going to be saved
+    string key;
+    string value;
+}TOTO;
+vector<TOTO> fin_tokens;
+void display (Token token); //display function 
 
 int main(int argc, char *argv[])
 {
-    std::cout << "argc == " << argc << '\n';
+    // Passing in the program file & modifiying it 
+    ifstream inFile;
+    inFile.open("test2.txt"); // Keep in mind program file directory 
+    stringstream strFile;
+    strFile << inFile.rdbuf();
+
+    string fileString = strFile.str(); // file is now a long string stored as fileString
+    replace( fileString.begin(), fileString.end(), '\n', ' ' ); // find newlines & replace with space
```

Comment:
This won't work when detecting single line comments because the lexer will need to read to the end of the line. You'll need to ignore newlines later in the lexing process.

Reply:
Will correct & update branch with it